src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -254,15 +255,22 @@ def check_exists(changed_headers, parser, values, proj, missing_headers, i):
             # Check if the next file is in missing_headers
             return check_exists(changed_headers, parser, values, proj, missing_headers, i + 1)
         else:
+            # Save current copy of files[i]
+            tempfile = NamedTemporaryFile(mode="w", delete=True).name
+            shutil.copyfile(files[i], tempfile)
+
```

Comment:
This might make the hook extremely slow... Questions:

* If I do ``pre-commit run``, the hook is only acting upon modified files, right?

Reply:
I believe so, I typically just run "pre-commit" to run on the committed files but I think you have to have --all-files to run on all of them